conans/client/toolchain/msbuild.py

```diff
@@ -0,0 +1,99 @@
+        runtime_library = {"MT": "MultiThreaded",
+                           "MTd": "MultiThreadedDebug",
+                           "MD": "MultiThreadedDLL",
+                           "MDd": "MultiThreadedDebugDLL"}.get(runtime, "")
+
+        content = textwrap.dedent("""\
+            <?xml version="1.0" encoding="utf-8"?>
+            <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+              <ItemDefinitionGroup>
+                <ClCompile>
+                  <PreprocessorDefinitions>
+                     {};%(PreprocessorDefinitions)
+                  </PreprocessorDefinitions>
+                  <RuntimeLibrary>{}</RuntimeLibrary>
+                  <LanguageStandard>{}</LanguageStandard>
+                </ClCompile>
+              </ItemDefinitionGroup>
+            </Project>
+            """)
+        definitions = ";".join([format_macro(k, v) for k, v in self.definitions.items()])
+        # It is useless to set PlatformToolset in the config file, because the conditional checks it
+        cppstd = "stdcpp%s" % cppstd if cppstd else ""
+        config_props = content.format(definitions, runtime_library, cppstd)
+        config_filepath = os.path.abspath(config_filename)
+        self._conanfile.output.info("MSBuildToolchain created %s" % config_filename)
+        save(config_filepath, config_props)
+
+    def _write_main_toolchain(self, config_filename, condition):
+        main_toolchain_path = os.path.abspath("conan_toolchain.props")
+        if os.path.isfile(main_toolchain_path):
+            content = load(main_toolchain_path)
+        else:
+            content = textwrap.dedent("""\
+                <?xml version="1.0" encoding="utf-8"?>
+                <Project ToolsVersion="4.0"
+                        xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+                    <ImportGroup Label="PropertySheets" >
+                    </ImportGroup>
+                </Project>
+                """)
+
+        dom = minidom.parseString(content)
+        import_group = dom.getElementsByTagName('ImportGroup')[0]
+        children = import_group.getElementsByTagName("Import")
+        for node in children:
+            if (config_filename == node.getAttribute("Project") and
+                    condition == node.getAttribute("Condition")):
+                break  # the import statement already exists
+        else:  # create a new import statement
+            import_node = dom.createElement('Import')
+            import_node.setAttribute('Condition', condition)
+            import_node.setAttribute('Project', config_filename)
+            import_group.appendChild(import_node)
+
+        conan_toolchain = dom.toprettyxml()
+        conan_toolchain = "\n".join(line for line in conan_toolchain.splitlines() if line.strip())
+        self._conanfile.output.info("MSBuildToolchain writing %s" % "conan_toolchain.props")
+        save(main_toolchain_path, conan_toolchain)
```

Comment:
not sure if needed, but we specified `utf-8` encoding in the file itself. maybe it makes sense to specify utf-8 encoding for `tools.save` as well

Reply:
utf-8 is already the default in ``save()``